tests/hypothesis_strategies/constraints.py

```diff
@@ -71,6 +74,106 @@ def discrete_excludes_constraints(
     return DiscreteExcludeConstraint(parameter_names, conditions, combiner)
 
 
+@st.composite
+def discrete_dependencies_constraints(
+    draw: st.DrawFn,
+    parameters: Optional[List[DiscreteParameter]] = None,
+    affected_parameter_names: Optional[List[List[str]]] = None,
+):
+    if parameters is None:
+        # Draw random unique parameter names
+        # If affected parameters are given the list length must be respected
+        parameter_names = draw(
```

Comment:
You have 3 checks regarding whether or not `affected_parameter_names` is `None`, I'd appreciate if this would be shortened

Reply:
they happen in different subcontexts and I dont think the assumption that this can be shortened is necessarily true (this is not a default args check that you can do once at the top of a function), I can look over it again but without a concrete suggestion I have no starting point

Reply:
I'd have proposed to just have something like `aff_param_exist = affected_parameter_names is None` and then change the check in lines 91, 95 and 98. But if you think this is unnecessary feel free to just resolve.

Reply:
no longer relevant since stuff very likely needs a refactor anyway once we come to the searchspaces (see other thread). Hence, will close